packages/salesforce-adapter/src/elements_url_retreiver/lightining_url_resolvers.ts

```diff
@@ -0,0 +1,174 @@
+}
+
+const metadataTypeResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  const internalId = getInternalId(element)
+  if (isType(element)
+    && element.annotations.apiName?.endsWith('__mdt')
+    && internalId !== undefined) {
+    return (new URL(`${baseUrl}lightning/setup/CustomMetadata/page?address=%2F${internalId}%3Fsetupid%3DCustomMetadata`))
+  }
+  return undefined
+}
+
+const objectResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  const typeIdentfier = getTypeIdentifier(element)
+  if (isObjectType(element)
+    && metadataType(element) === 'CustomObject'
+    && typeIdentfier !== undefined) {
+    return new URL(`${baseUrl}lightning/setup/ObjectManager/${typeIdentfier}/Details/view`)
+  }
+  return undefined
+}
+
+const fieldResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  if (isField(element)) {
+    const fieldIdentifier = getFieldIdentifier(element)
+    const typeIdentfier = getTypeIdentifier(element.parent)
+    if (fieldIdentifier !== undefined
+      && metadataType(element.parent) === 'CustomObject'
+      && typeIdentfier !== undefined) {
+      return new URL(`${baseUrl}lightning/setup/ObjectManager/${typeIdentfier}/FieldsAndRelationships/${fieldIdentifier}/view`)
+    }
+  }
+  return undefined
+}
+
+const flowResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  const internalId = getInternalId(element)
+  if (isInstanceElement(element)
+    && metadataType(element) === 'Flow'
+    && element.value.processType === 'Flow'
+    && internalId !== undefined) {
+    return (new URL(`${baseUrl}builder_platform_interaction/flowBuilder.app?flowId=${internalId}`))
+  }
+  return undefined
+}
+
+const workflowResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  if (isInstanceElement(element)
+    && metadataType(element) === 'Flow'
+    && element.value.processType === 'Workflow') {
+    // It seems all the process builder flows has the same url so we return the process buider home
+    return new URL(`${baseUrl}lightning/setup/ProcessAutomation/home`)
+  }
+  return undefined
+}
+
+const queueResolver: UrlResolver = async (element, baseUrl, _elementIDResolver) => {
+  const internalId = getInternalId(element)
+  if (isInstanceElement(element)
+    && metadataType(element) === 'Queue'
+    && getInternalId(element) !== undefined) {
+    return new URL(`${baseUrl}lightning/setup/Queues/page?address=%2Fp%2Fown%2FQueue%2Fd%3Fid%3D${internalId}`)
+  }
+  return undefined
+}
+
+const layoutResolver: UrlResolver = async (element, baseUrl, elementIDResolver) => {
+  const internalId = getInternalId(element)
+  const [parentRef] = getParents(element)
+
+  if (isInstanceElement(element)
+      && metadataType(element) === 'Layout'
+      && internalId !== undefined
+      && isReferenceExpression(parentRef)) {
+    const parent = await elementIDResolver(element.annotations[CORE_ANNOTATIONS.PARENT][0].elemId)
```

Comment:
the point of checking `parentRef` was that you should use the thing you check
this isn't helpful when the code is still using something that was not checked here...
```suggestion
    const parent = await elementIDResolver(parentRef.elemId)
```

Reply:
Oops